src/sidebar/services/test/streamer-test.js

```diff
@@ -89,9 +92,22 @@ describe('StreamerService', () => {
       fakeAuth,
       fakeGroups,
       fakeSession,
+      { setTimeout: fakeSetTimeout },
     );
   }
 
+  function timeoutAsPromise() {
+    const { resolve, promise } = promiseWithResolvers();
+    fakeSetTimeout.callsFake(callback =>
+      setTimeout(() => {
+        callback();
+        resolve();
+      }, 1),
+    );
+
+    return promise;
+  }
```

Comment:
I want to think on a way to extract this as a reusable utility, as I have defined a similar function in at least 3 tests already. The "problem" is that depends on a local variable, that's why I have been just copy-pasting it.

Probably the simplest solution would be to simply pass the `fakeSetTimeout` as argument, then it would be fully reusable.